Add dev profile to strip symbols and disable debug info (ref #3610)

Nutomic opened the pull request:
This significantly speeds up builds:
- with strip symbols and debug 0: 43s
- without: 169s

phiresky commented:
Would be good to try with just debug=0 because those symbols are really useful for stack traces (e.g. anyhow errors, the use of tracing spans isn't consistent everywhere) and i think they are stripped only after the compilation process anyways?

also a comment would be good referencing the issue so no one tries to remove it again like i would

Nutomic commented:
Okay updated the measurements. Somehow Im still getting inconsistent results with the same build command taking sometimes 2 minutes and then 5 minutes for no reason. Anyway it looks like only debug 0 is needed, so I updated it like that and added a comment.
